homeassistant/components/daikin/sensor.py

```diff
@@ -179,11 +214,18 @@ def __init__(
         """Initialize the sensor."""
         self.entity_description = description
         self._api = api
-        self._attr_name = f"{api.name} {description.name}"
+        self._attr_entity_registry_enabled_default = description.enabled_default
+        if description.cross_device:
+            self._attr_name = f"Daikin {description.name}"
+        else:
+            self._attr_name = f"{api.name} {description.name}"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID."""
+        if self.entity_description.cross_device:
+            # Cross-device sensors should be generated once for all devices
```

Comment:
How is this controlled?

Reply:
What do you mean by "controlled"? By given a non-unique id to those entities, they are created only once even if you create multiple daikin integrations

Reply:
That's not how unique_id should be used. The integration needs to take care not to create duplicate entities with the same unique id in the integration. There will be an error otherwise.

Reply:
The error was expected indeed.

Should I avoid the error, would you have an example of an existing integration checking for existing entities before creating new ones? I imagine this logic should take place in `async_setup_entry` by using the `hass` object but I was not able to find out how yet

Reply:
What's the topology of the devices captured by this measurement? If a user connects to more than one controller, if more than one config entry, how will this measurement be affected?

Reply:
An AC installation may have multiple indoor devices plugged on a single outdoor one. By connecting to one indoor Daikin AC through Wifi, the user will be able to interact with the indoor device (control mode, indoor temp, fan speed, device energy, etc.) but also outdoor device's sensors (outdoor temp, compressor frequency, whole installation energy, etc.)

If a user connects to more than one controller, each controller will thus report exactly the same values with a small time shift due to the minimum refresh rate of each entity. IMHO there is no point of creating those entities multiple times, it can only create confusion (e.g: "Each controller is reporting the compressor frequency, so the compressor must certainly be in the indoor devices")

Reply:
Won't it be confusing if one config entry is missing the common sensors while another config entry has them? How would the user know where to look?

Reply:
I would still have the idea to have a outdoor device with these entities. Some custom integrations do this for the main services/devices as well, e.g. the Home Connect HACS integration version. There you have the n devices and on top another device. But I'm not the expert, if this is possible here or allowed at all. But would seem to be the most logical option.

Reply:
> Won't it be confusing if one config entry is missing the common sensors

It could indeed. I don't think it's feasible but it'd be great if the config flow was able to create two config entries at once for the same IP, wdyt? One for the indoor device and one for the outdoor one if it does not already exist.

Otherwise we would also create the outdoor sensors on all config entries but enable them somehow only in one config entry 🤔

Reply:
Oh actually from the "indoor config entry" setup I could trigger another async config flow to create the "outdoor config entry" ([example](https://github.com/home-assistant/core/blob/0.68.0/homeassistant/components/hue/__init__.py#L104))

I'll try this

Reply:
The automatic creation of another unique config entry for the outdoor device worked but I finally found clearer to distinguish the indoor/outdoor sensors by their associated device: each config entry will create the outdoor sensors (unless the "Enable outdoor sensors" option is disabled during the config flow) which will be associated to an "Outdoor" device instead.

Reply:
Isn't it possible that the user has more than one outdoor device and multiple indoor devices connected to each of the outdoor devices in the same home? How do we handle this in the integration?

Reply:
I've not seen that in the API. The pydaikin lib does not support such a scenario.

Reply:
What happens if the user tries to connect more than one outdoor device?

Reply:
I've never seen an installation plugged on two outdoor devices at once, but that could indeed happen if the user has bought two independent Daikin installations in its house, which will be the case of only a few people. Anyway, if it happens, all sensors related to both outdoor devices will be created in HA, but will be linked to the same "outdoor device". The user will be free to disable or not each sensor.

I don't think the current Wifi adapters give any kind of unique identifiers for the outdoor device; the only field that I don't understand yet and could be an identifier is the `rev` one, equal to `203DE8C` in my case. @fredrike do you know something about this field? Maybe a serial number? I haven't been able to find info on the net

Reply:
I have thought about that issue a lot and I think we could generate an outdoor unit's UUID by using the monthly energy consumption of the last two years which is recorded by the outdoor unit: we're talking about 13 to 24 integers, that should be enough to ensure uniqueness, but this ID would change each month (is that an issue?). And if someone have two outdoor units with none of them supporting energy monitoring, we would fall back to a default ID but still create one outdoor sensor per outdoor unit

Wdyt?